api/v1alpha2/conversion_test.go

```diff
@@ -85,5 +153,93 @@ func TestConvertMachine(t *testing.T) {
 			g.Expect(dst.GetAnnotations()[utilconversion.DataAnnotation]).To(ContainSubstring("secret-data"))
 		})
 	})
+}
+
+func TestConvertMachineSet(t *testing.T) {
+	g := NewWithT(t)
+
+	t.Run("to hub", func(t *testing.T) {
+		t.Run("should convert the Spec.ClusterName from label", func(t *testing.T) {
+			src := &MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						MachineClusterLabelName: "test-cluster",
+					},
+				},
+			}
+			dst := &v1alpha3.MachineSet{}
+
+			g.Expect(src.ConvertTo(dst)).To(Succeed())
+			g.Expect(dst.Spec.ClusterName).To(Equal("test-cluster"))
+		})
+	})
+
+	t.Run("from hub", func(t *testing.T) {
+		t.Run("preserves the hub version", func(t *testing.T) {
+			src := &v1alpha3.MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "hub",
+				},
+			}
+			dst := &MachineSet{}
+
+			g.Expect(dst.ConvertFrom(src)).To(Succeed())
+			g.Expect(dst.GetAnnotations()[utilconversion.DataAnnotation]).To(ContainSubstring("hub"))
+		})
+	})
+}
+
+func TestConvertMachineDeployment(t *testing.T) {
+	g := NewWithT(t)
+
+	t.Run("to hub", func(t *testing.T) {
+		t.Run("should convert the Spec.ClusterName from label", func(t *testing.T) {
+			src := &MachineDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						MachineClusterLabelName: "test-cluster",
+					},
+				},
+			}
+			dst := &v1alpha3.MachineDeployment{}
+
+			g.Expect(src.ConvertTo(dst)).To(Succeed())
+			g.Expect(dst.Spec.ClusterName).To(Equal("test-cluster"))
+		})
+	})
 
+	t.Run("from hub", func(t *testing.T) {
+		t.Run("preserves the hub version", func(t *testing.T) {
+			src := &v1alpha3.MachineDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "hub",
+				},
+			}
+			dst := &MachineDeployment{}
+
+			g.Expect(dst.ConvertFrom(src)).To(Succeed())
+			g.Expect(dst.GetAnnotations()[utilconversion.DataAnnotation]).To(ContainSubstring("hub"))
+		})
+
+		t.Run("preserves Spec.Paused", func(t *testing.T) {
+			src := &v1alpha3.MachineDeployment{
+				Spec: v1alpha3.MachineDeploymentSpec{
+					Paused: true,
+				},
+			}
+			dst := &MachineDeployment{}
+
+			g.Expect(dst.ConvertFrom(src)).To(Succeed())
+			g.Expect(dst.GetAnnotations()[utilconversion.DataAnnotation]).To(ContainSubstring("paused\":true"))
+		})
+
+		t.Run("preserves Status.Phase", func(t *testing.T) {
+			src := &v1alpha3.MachineDeployment{}
+			src.Status.SetTypedPhase(v1alpha3.MachineDeploymentPhaseRunning)
+			dst := &MachineDeployment{}
+
+			g.Expect(dst.ConvertFrom(src)).To(Succeed())
+			g.Expect(dst.GetAnnotations()[utilconversion.DataAnnotation]).To(ContainSubstring("phase\":\"Running\""))
```

Comment:
I think I'd rather (or additionally) see a call to ConvertTo and then verify that the new v1a3 object has the expected fields set (could do this in a single test instead of one per field)